File: examples/corpus/ts/03_cache.after.ts
class Cache {
  private store: Record<string, any> = {};
  get(key: string): any {
    return this.store[key];
  }
  set(key: string, value: any): void {
    this.store[key] = value;
  }
}

function merge(base: Record<string, any>, other: Record<string, any>): Record<string, any> {
  return { ...base, ...other };
}

File: examples/corpus/go/03_cache.after.go
package main

type Cache struct {
	store map[string]interface{}
}

func (c Cache) Get(key string) interface{} {
	return c.store[key]
}

func (c Cache) Set(key string, value interface{}) {
	c.store[key] = value
}

func merge(base, other map[string]interface{}) map[string]interface{} {
	out := map[string]interface{}{}
	for k, v := range base {
		out[k] = v
	}
	for k, v := range other {
		out[k] = v
	}
	return out
}
